keps/sig-node/0035-20190130-topology-manager.md

```diff
@@ -446,22 +449,31 @@ _Figure: Topology Manager fetches affinity from hint providers._
 
 ### Deprecates the Policy Flag of Topology Manager
 Since the Pod Spec is able to specify preferred topology policy of a Pod
-and the Topology Manager starts to see `v1core.PodSpec.Topology.Policy` field when it is configured with the dynamic policy,
-the policy flag will be deprecated on Beta stage(v1.20) and the danamic policy will be a default behavior of the Topology Manager.
+and the Topology Manager is able to read `v1core.PodSpec.Topology.Policy` field when it is configured with the dynamic policy,
+the policy flag will be deprecated on Beta stage(v1.20) and the dynamic policy will be a default behavior of the Topology Manager.
 
 Before the deprecation of the policy flag,
-the dynamic policy should be configured on a node where operator expects that the Topology Manager will see a Pod Spec to get topology policy of a pod.
+the dynamic policy should be configured on a node where operator expects the Topology Manager to get a topology policy of a pod from a Pod Spec.
 
-If the other policy flags are configured on a node than the dynamic policy flag, the cofigured policy flags take precedence over the topology policy of Pod Spec.
-The behavior of Topology policies is listed in the below.
+If topology policy is configured on the node as none/best-effort/restriced/single-numa-node, it takes precedence over the topology policy specified in Pod Spec.
+The behavior of Topology policies is listed below.
 - **none(default)**
-  * Kubelet does not consult Topology Manager for placement decisions.
+  * Kubelet does not consult with Topology Manager for placement decisions.
 - **best-effort/restricted/single-numa-node**
-  * If a node is configured with these policies, the Topology Manager will coordinates resource assignment based on configured topology policy for all the pod of a node.
-  * The Topology Manager will not see the `v1core.PodSpec.Topology.Policy` field of Pod Spec.
+  * If a node is configured with these policies, the Topology Manager will coordinate assignment of resources based on configured topology policy for all pods that were scheduled to the node.
+  * The Topology Manager will not read the `v1core.PodSpec.Topology.Policy` field of Pod Spec.
 - **dynamic**
-  * The Topology Manager see `v1core.PodSpec.Topology.Policy` field to coordinates resource assignment of a Pod.
+  * The Topology Manager reads `v1core.PodSpec.Topology.Policy` field to coordinate assignment of resources for a Pod.
 
+## Practical challenges
+Non-NUMA aware scheduler may bind a pod to a node, where topology requirements cannot be satisfied. Topology Manager will reject a pod admission with TopologyError and pod will result in a `Terminated` state on a node.
```

Comment:
that problem exists in current implementation with per node defined policies.

Reply:
Technically, that problem originated from the concept of pod and its lifecycle.
`In general, Pods remain until a human or controller process explicitly removes them.`
https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-lifetime